### PR TITLE
do not set build user context

### LINKED ISSUE #3995: Incorrect build context in config-reloader image logs
**What happened?**
Started prometheus config reloader and second log line indicates build context which is incorrect:
```
level=info ts=2021-04-16T08:39:05.36818227Z caller=main.go:148 build_context="(go=go1.16.3, user=simonpasquier, date=20210413-15:46:43)"
```

**Did you expect to see something different?**
I expected to see build context which comes from CI, possibly sth like `root@....`

**How to reproduce it (as minimally and precisely as possible)**:
Start config-reloader from quay.io

**Environment**

* Prometheus Operator version: 0.47.0

* Kubernetes version information: Not relevant

* Kubernetes cluster kind: Not relevant

* Manifests: Not relevant

* Prometheus Operator Logs:

```
Not relevant
```

**Anything else we need to know?**:

This was spotted in https://github.com/prometheus-operator/kube-prometheus/issues/1099 issue.

CC @simonpasquier

### PR DESCRIPTION
**Pull request by paulfantom**
Fixes #3995

**Release Note Template (will be copied)**

```release-note:enhancement
Set proper build context in version package
```

### CODE REVIEW

**Comment by paulfantom** [comment]
Local run shows:
```
./prometheus-config-reloader 
level=info ts=2021-05-05T14:17:39.580146604Z caller=main.go:147 msg="Starting prometheus-config-reloader" version="(version=0.47.1, branch=, revision=)"
level=info ts=2021-05-05T14:17:39.580478194Z caller=main.go:148 build_context="(go=go1.16, user=, date=20210505-15:54:10)"
level=info ts=2021-05-05T14:17:39.58068246Z caller=reloader.go:183 msg="nothing to be watched"
```

**Comment by simonpasquier** [comment]
maybe we could pass a value indicating that it's built from the CI pipeline? Also could the `Makefile` not use environment variables specific to GitHub actions but rather env vars that could be set externally?

**Comment by paulfantom** [comment]
Rebased, updated to include suggestions, and fixed few smaller issues.

File: Makefile [hunk 1]
```diff
@@ -43,23 +43,35 @@ K8S_GEN_DEPS+=$(TYPES_V1_TARGET)
...
 K8S_GEN_DEPS+=$(foreach bin,$(K8S_GEN_BINARIES),$(TOOLS_BIN_DIR)/$(bin))
 
+BUILD_DATE=$(shell date +"%Y%m%d-%T")
+# source: https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables
+ifndef GITHUB_ACTIONS
+	BUILD_USER?=$(USER)
+	BUILD_BRANCH?=$(shell git branch --show-current)
+	BUILD_REVISION?=$(shell git rev-parse --short HEAD)
+else
+	BUILD_USER=Action-Run-ID-$(GITHUB_RUN_ID)
```
Comment (simonpasquier) [inline Makefile:53]: :+1:

**Review by simonpasquier — COMMENTED** [review]
lgtm

**Pull request merged**